streams/common/src/main/java/com/hortonworks/streamline/streams/common/StreamlineEventImpl.java

```diff
@@ -283,9 +244,13 @@ public StreamlineEvent addFieldAndValue(String key, Object value) {
      */
     @Override
     public StreamlineEvent addHeaders(Map<String, Object> headers) {
-        StreamlineEventImpl result = new StreamlineEventImpl(this);
-        result.header.putAll(headers);
-        return result;
+        Map<String, Object> headerMap = new HashMap<>();
```

Comment:
Use an immutable map here so the build below can just use it and avoid the extra HashMap?

Reply:
Same here.